<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="OpenGLPanel3D" name="GLPanel3D" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>681</width>
      <height>681</height>
     </rect>
    </property>
   </widget>
   <widget class="OpenGLPanel2D" name="GLPanel2D" native="true">
    <property name="geometry">
     <rect>
      <x>709</x>
      <y>50</y>
      <width>511</width>
      <height>501</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="jitterButton">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>10</y>
      <width>51</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Jitter</string>
    </property>
   </widget>
   <widget class="QSlider" name="weightSumSlider">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>10</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>50</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="computeHeightButton">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>10</y>
      <width>131</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Compute Heights</string>
    </property>
   </widget>
   <widget class="QPushButton" name="iterateButton">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>10</y>
      <width>61</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Iterate</string>
    </property>
   </widget>
   <widget class="QPushButton" name="subdivideButton">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>10</y>
      <width>97</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Subdivide</string>
    </property>
   </widget>
   <widget class="QPushButton" name="resetButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>51</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset</string>
    </property>
   </widget>
   <widget class="QPushButton" name="projectButton">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>10</y>
      <width>61</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Project</string>
    </property>
   </widget>
   <widget class="QPushButton" name="screenshotButton">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>10</y>
      <width>97</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Screenshot</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="editModeBox">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>560</y>
      <width>171</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Edit Mode</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <widget class="QRadioButton" name="cameraButton">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>116</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Camera</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="freeformHandleButton">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>161</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Handle (freeform)</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="heightHandleButton">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>141</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Handle (height)</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="deleteFaceButton">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>80</y>
       <width>116</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Delete Faces</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="renderOptionsBox">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>560</y>
      <width>221</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Render Options</string>
    </property>
    <widget class="QCheckBox" name="referenceCheckBox">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>191</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Show reference surface</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="networkCheckBox">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>191</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Show thrust network</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="networkSurfaceCheckBox">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>60</y>
       <width>231</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Show thrust network surface</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="residualLabel">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>710</y>
      <width>231</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Current thrust network residual:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="residualEdit">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>710</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_Mesh"/>
    <addaction name="actionLoad_Mesh"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionCenter_Cameras"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionLoad_Mesh">
   <property name="text">
    <string>Load Mesh</string>
   </property>
  </action>
  <action name="actionNew_Mesh">
   <property name="text">
    <string>New Mesh</string>
   </property>
  </action>
  <action name="actionCenter_Cameras">
   <property name="text">
    <string>Center Cameras</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OpenGLPanel3D</class>
   <extends>QWidget</extends>
   <header>openglpanel3d.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>OpenGLPanel2D</class>
   <extends>QWidget</extends>
   <header>openglpanel2d.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>502</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
